### PR TITLE
Add key backup modal

### PR DESCRIPTION
**Pull request by chartgerink**
This PR adds a modal that forces users to back up their key. This is based on the experience that users did not follow on the welcome tour instructions and ended up not creating a backup of their private key. The implemented design is as follows (illustration provided by @psobrakseaton):

Screenshot 2021-02-05 at 10 21 23

Note: This modal does not yet appear for existing users of Hypergraph. There is some commented out code in the pull request that aimed to implement this, but alas, I ran into a wall with how the modals were set up. I ended up paginating to the extent that it hit `if (!dialogs[page]) return null` which logically led to a black screen upon close. Any help appreciated to make existing users also back up their keys :)

### CODE REVIEW

**Comment by juliangruber** [comment]
> There is some commented out code in the pull request that aimed to implement this, but alas, I ran into a wall with how the modals were set up. I

Yeah, showing just one slide wasn't considered with the initial design.

**Comment by juliangruber** [comment]
> Any help appreciated to make existing users also back up their keys :)

You could show a `Notification` and on click do the same thing as the `Copy your key` button?

File: app/index.js [hunk 1]
```diff
@@ -138,10 +138,12 @@ const App = () => {
   )
   const [isFinding, setIsFinding] = useState(Boolean(findModalUrl))
   const [showWelcome, setShowWelcome] = useState()
+  // const [keyCopied, setKeyCopied] = useState()
```
Comment (juliangruber) [inline app/index.js:141]: is this still a WIP / should it be a draft PR?

File: index.js [hunk 2]
```diff
@@ -100,6 +102,8 @@ const updateMenu = () => {
                 const zip = new AdmZip()
                 zip.addLocalFolder(p2pcommonsDir)
                 zip.writeZip(filePath)
+
+                store.set('keyCopied', true)
```
Comment (juliangruber) [inline index.js:106]: `keyBackedUp` would be more semantic

Comment (chartgerink) [reply index.js]: Thanks! I changed it. I am still getting the hang of semantic naming and your guidance helps ☺️

File: app/index.js (hunk 1 above)
Comment (chartgerink) [reply app/index.js]: I was sloppy. 😅  Removed it.

File: app/components/welcome/welcome.js [hunk 3]
```diff
@@ -17,9 +18,10 @@ import Avatar from '../avatar/avatar'
 import { archiveModule } from '../../lib/vault'
 import Anchor from '../anchor'
 import { ProfileContext } from '../../lib/context'
-import { ipcRenderer } from 'electron'
+import { remote, ipcRenderer } from 'electron'
 import TermsOfUse from './terms-of-use'
 import { productName } from '../../../package'
+const AdmZip = require('adm-zip')
```
Comment (juliangruber) [inline app/components/welcome/welcome.js:24]: this file use `import` syntax

File: app/components/welcome/welcome.js [hunk 4]
```diff
@@ -313,9 +367,9 @@ const dialogs = [
   }
 ]
 
-const Welcome = ({ p2p, setProfileUrl }) => {
+const Welcome = ({ p2p, start, setProfileUrl }) => {
   const { url: profileUrl } = useContext(ProfileContext)
-  const [page, setPage] = useState(profileUrl ? 1 : 0)
+  const [page, setPage] = useState(start || (profileUrl ? 1 : 0))
```
Comment (juliangruber) [inline app/components/welcome/welcome.js:372]: the `start` option seems not to be used

File: app/index.js [hunk 5]
```diff
@@ -191,16 +191,13 @@ const App = () => {
...
-  }
-
-  return (
+  return loading ? (
+    <Container />
+  ) : !profileUrl || showWelcome ? (
+    <Container profileUrl={profileUrl}>
+      <Welcome p2p={p2p} setProfileUrl={setProfileUrl} />
+    </Container>
+  ) : (
```
Comment (juliangruber) [inline app/index.js:200]: nice!

File: index.js [hunk 6]
```diff
@@ -235,6 +247,28 @@ const main = async () => {
   mainWindow = await createMainWindow()
   app.setAsDefaultProtocolClient('hypergraph')
   if (app.isPackaged) autoUpdater.checkForUpdatesAndNotify()
+  if (!store.get('keyBackedUp')) {
+    const backUpKey = new Notification({
+      title: 'Back up your key',
+      body:
+        'To ensure you keep access to your account, back up your key now. Click here.'
```
Comment (juliangruber) [inline index.js:254]: This doesn't read friendly to me. What about something like

> To ensure you keep access to your account, back up your key by clicking here.

**Review by juliangruber — CHANGES_REQUESTED** [review]

**Pull request merged**